var/spack/repos/builtin/packages/casacore/package.py

```diff
@@ -22,6 +23,8 @@ class Casacore(CMakePackage):
     version('3.0.0', sha256='6f0e68fd77b5c96299f7583a03a53a90980ec347bff9dfb4c0abb0e2933e6bcb')
     version('2.4.1', sha256='58eccc875053b2c6fe44fe53b6463030ef169597ec29926936f18d27b5087d63')
 
+    depends_on('cmake@3.7.1:', type='build')
```

Comment:
Does this dependency apply to all available versions of this package?

Reply:
Yes, it should. Although the CMakeLists.txt file of the oldest (packaged) casacore version has a minimum CMake version of 2.6.0, there are comments indicating that versions 3.x should also work. To check, I just built the oldest casacore version with CMake 3.16.3, and verified that the proper HDF5 libraries are linked by the casacore libraries. I've also verified that the change does what's intended for the newest casacore package, too. I hope that's sufficient, but I could try building every packaged version, if that necessary.